### PR TITLE
coverage for `captureOrderTransaction`

### PR DESCRIPTION
**Pull request by tiutiunnykleasoft**
* Implemented coverage for `captureOrderTransaction` method from `ginger-php`.
* Implemented `OrderStub` to simplify valid order creation methods.
* Implemented `CaptureOrderTransactionTest`.
* Implemented `OrderNotFoundException` and used in `getOrder` method.
* Implemented `CaptureFailedExcpetion`.
* Implemented `getStatus` and `getId` methods to Order entity.
* Refactored `sendOrder` to return `Order` entity instead of array.
* Refactored code according to `PHPStan` analyse.
* Refactored `AbstractCollection` to not include `class-string` into construct.
* Added `PHPStan` to `composer.json` to a dev environment.

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -544,6 +544,65 @@
             ],
             "time": "2021-12-08T12:19:24+00:00"
         },
+        {
+            "name": "phpstan/phpstan",
+            "version": "1.7.10",
+            "source": {
+                "type": "git",
+                "url": "https://github.com/phpstan/phpstan.git",
```
Comment (BohdanLeasoft) [inline composer.lock:552]: Should composer.lock file be in the repository?

File: src/Entities/Line.php [hunk 2]
```diff
@@ -44,7 +44,7 @@ public function __construct(
         string            $merchantOrderLineId,
         string            $name,
         int               $quantity,
-        float             $amount,
+        float              $amount,
         int               $vatPercentage,
         private ?Currency $currency = null,
```
Comment (BohdanLeasoft) [inline src/Entities/Line.php:49]: Looks like here is an extra space
 ` float              $amount,`

**Review by Denis-Tsyganok-Leasoft — APPROVED** [review]

File: src/Entities/Transaction.php [hunk 3]
```diff
@@ -122,6 +144,48 @@ enum: [
...
+
+        if ($mscVatClass) $this->mscVatClass = $this->createSimpleField(
+            propertyName: "msc_vat_class",
+            value: $mscVatClass
+        );
+
+        if ($mscVatAmount) $this->mscVatAmount = $this->createSimpleField(
+            propertyName: 'msc_vat_amount',
+            value: $mscVatAmount
+        );
```
Comment (BohdanLeasoft) [inline src/Entities/Transaction.php:170]: It is not matter, but propertyNames in single quotes. Like 'msc_vat_percentage'
But on line 163:` propertyName: "msc_vat_class"` in double quotes

**Review by BohdanLeasoft — APPROVED** [review]
Looks nice :+1:

File: composer.lock (hunk 1 above)
Comment (tiutiunnykleasoft) [reply composer.lock:552]: Yes

File: src/Entities/Line.php (hunk 2 above)
Comment (tiutiunnykleasoft) [reply src/Entities/Line.php:49]: In my IDE there is no additionall space, idk, maybe bug

File: src/Entities/Transaction.php (hunk 3 above)
Comment (tiutiunnykleasoft) [reply src/Entities/Transaction.php:170]: OK

**Pull request merged**